pykafka/rdkafka/simple_consumer.py

```diff
@@ -0,0 +1,240 @@
+
+    For an overview of how configuration keys are mapped to librdkafka's, see
+    _mk_rdkafka_config_lists.
+    """
+    def __init__(self,
+                 topic,
+                 cluster,
+                 consumer_group=None,
+                 partitions=None,
+                 fetch_message_max_bytes=1024 * 1024,
+                 num_consumer_fetchers=1,
+                 auto_commit_enable=False,
+                 auto_commit_interval_ms=60 * 1000,
+                 queued_max_messages=10**5,  # NB differs from SimpleConsumer
+                 fetch_min_bytes=1,
+                 fetch_wait_max_ms=100,
+                 offsets_channel_backoff_ms=1000,
+                 offsets_commit_max_retries=5,
+                 auto_offset_reset=OffsetType.EARLIEST,
+                 consumer_timeout_ms=-1,
+                 auto_start=True,
+                 reset_offset_on_start=False):
+        callargs = {k: v for k, v in vars().items()
+                         if k not in ("self", "__class__")}
+        self._rdk_consumer = None
+        self._poller_thread = None
+        self._stop_poller_thread = cluster.handler.Event()
+        # super() must come last for the case where auto_start=True
+        super(RdKafkaSimpleConsumer, self).__init__(**callargs)
+
+    def _setup_fetch_workers(self):
+        brokers = b','.join(b.host + b":" + get_bytes(str(b.port))
+                           for b in self._cluster.brokers.values())
+        partition_ids = list(self._partitions_by_id.keys())
+        start_offsets = [
+            self._partitions_by_id[p].next_offset for p in partition_ids]
+        conf, topic_conf = self._mk_rdkafka_config_lists()
+
+        self._rdk_consumer = _rd_kafka.Consumer()
+        log.debug("Configuring _rdk_consumer...")
+        self._rdk_consumer.configure(conf=conf)
+        self._rdk_consumer.configure(topic_conf=topic_conf)
+
+        start_kwargs = {"brokers": brokers,
+                        "topic_name": self._topic.name,
+                        "partition_ids": partition_ids,
+                        "start_offsets": start_offsets}
+        log.debug("Starting _rdk_consumer with {}".format(start_kwargs))
+        self._rdk_consumer.start(**start_kwargs)
+
+        # Poll: for a consumer, the main reason to poll the handle is that
+        # this de-queues log messages at error level that might otherwise be
+        # held up in librdkafka
+        def poll(rdk_handle, stop_event):
+            while not stop_event.is_set():
+                rdk_handle.poll(timeout_ms=1000)
+            log.debug("Exiting RdKafkaSimpleConsumer poller thread.")
+
+        self._stop_poller_thread.clear()
+        self._poller_thread = self._cluster.handler.spawn(
+            poll, args=(self._rdk_consumer, self._stop_poller_thread))
+
+    def consume(self, block=True):
+        timeout_ms = self._consumer_timeout_ms if block else 1
+        try:
+            msg = self._rdk_consumer.consume(timeout_ms)
```

Comment:
Argh, thanks, you're right, that's not very user friendly. If I put in a log line where you said it still blocks for me (unless I'm very quick). I think I'll just add another level of looping, so that we always call the C function with `timeout_ms=500`, or something like that, and the outer loop checks for the real timeout. Coming up.

Re the iteration: when I try it here, it behaves exactly as you say it should? If I leave the default `consumer_timeout_ms=-1` it blocks forever at the end of the topic, and if I put it to something like `5000` I indeed see it block 5 seconds at the end of the topic.